src/enum.cr

```diff
@@ -139,10 +139,11 @@ struct Enum
     {% if @type.has_attribute?("Flags") %}
       String.build { |io| to_s(io) }
     {% else %}
-      case value
-      {% for member in @type.constants %}
-      when {{@type}}::{{member}}.value
```

Comment:
Why can't we just use `when {{member}}.value` here?
Also, is there a good reason why `Foo::FOO` isn't accessible from within the type? Isn't that the real bug?

Reply:
`undefined constant FOO`... but that should probably work.

Reply:
The problem is that type lookup doesn't take into account where the type lookup starts from. So trying to access `Foo::FOO` from anywhere will fail. Similar to `private` methods, you can only reference them without qualifying them.

That said, we can try to fix it...

The problem is, what happens if you have `Foo::Bar::Baz::Qux` and `Bar` is a private type inside `Foo`. Trying to find `Bar` inside `Foo` will fail. It'll need a big hack for it to work.

Reply:
`{{member}}` doesn't work because the enum member will be looked up relative to `Enum`, not the actual enum type. This is how type lookup works (constants can't be found in subtypes).

It's all a big mess... probably because we never spec'd it. I think the current workaround is fine. The issues with `private` seems to only affect this enum macro code.

Reply:
Oh, I get it. So this method actually should be generated within the subclass (the enum)?
Maybe using the no so well accepted `macro inherited`

Reply:
Anyway, I shouldn't diverge from the issue here. Just I wasn't aware of this issue. Resolving this conversation...